FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp

```diff
@@ -1367,14 +1367,16 @@ void OpDispatchBuilder::XCHGOp(OpcodeArgs) {
     // So x86-64 spec mandates this special case that even though it is a 32bit instruction and
     // is supposed to zext the result, it is a true no-op
     if (Op->Flags & FEXCore::X86Tables::DecodeFlags::FLAG_REP_PREFIX) {
-      // If this instruction has a REP prefix then this is architectually defined to be a `PAUSE` instruction.
-      // On older processors this ends up being a true `REP NOP` which is why they stuck this here.
+      // If this instruction has a REP prefix then this is architecturally
+      // defined to be a `PAUSE` instruction. On older processors this ends up
+      // being a true `REP NOP` which is why they stuck this here.
       _Yield();
     }
     return;
   }
 
-  OrderedNode *Src = LoadSource(GPRClass, Op, Op->Src[0], Op->Flags);
+  OrderedNode *Src = LoadSource(GPRClass, Op, Op->Src[0], Op->Flags,
+                                {.AllowUpperGarbage = true});
```

Comment:
Usually I try to justify with a comment or assert why garbage is ok ("it will be masked away by the following And" or whatever)

Reply:
Very good point!

Reply:
The correctness of this patch relies on `swpalb` ignoring the upper 24-bits of its source, etc. (Carefully look at the instcountci diff :smile: )